File: SampleApp_02_AspectInAndroidLib_App/src/main/java/me/leolin/sample02/app/MainActivity.java
package me.leolin.sample02.app;

import android.os.Bundle;
import android.widget.TextView;

import androidx.appcompat.app.AppCompatActivity;

import me.leolin.sample02.lib.MyService;


public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        MyService myService = new MyService();

        TextView textView = findViewById(R.id.textView);
        textView.setText(myService.getHelloString());
    }

}

File: SampleApp_00_AspectWithKotlin/src/main/java/th/co/digio/sampleapp_00_aspectwithkotlin/MainActivity.kt
package th.co.digio.sampleapp_00_aspectwithkotlin

import android.os.Bundle
import android.util.Log
import androidx.appcompat.app.AppCompatActivity
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        btn_main.setOnClickListener {
            Log.i("call", "from ${this.localClassName}")
        }
    }
}
